common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -3,48 +3,53 @@ import './ConfirmationModal.scss';
 import React from 'react';
 import translate from 'translations';
 import Big from 'bignumber.js';
+import { connect } from 'react-redux';
 import BaseWallet from 'libs/wallet/base';
-import { toUnit } from 'libs/units';
+import { toUnit, toTokenDisplay } from 'libs/units';
 import type { NodeConfig } from 'config/data';
 import type { RawTransaction } from 'libs/transaction';
+import type { Token } from 'config/data';
+import ERC20 from 'libs/erc20';
+import { getTokens } from 'selectors/wallet';
 
 import Modal from 'components/ui/Modal';
 import Identicon from 'components/ui/Identicon';
 
-// TODO: Handle other token types?
 type Props = {
+  // TODO: Use signed transaction, not raw transaction.
   rawTransaction: RawTransaction,
   wallet: BaseWallet,
   node: NodeConfig,
+  token: ?Token,
   onConfirm: RawTransaction => void,
   onCancel: () => void
 };
 
 type State = {
-  address: string,
-  value: string,
-  gasPrice: string,
-  nonce: string,
+  fromAddress: string,
   timeToRead: number
 };
 
-export default class ConfirmationModal extends React.Component {
+class ConfirmationModal extends React.Component {
   props: Props;
   state: State;
 
   constructor(props: Props) {
     super(props);
-    const { value, gasPrice, nonce } = props.rawTransaction;
 
     this.state = {
-      address: '',
-      value: toUnit(new Big(value, 16), 'wei', 'ether').toString(),
-      gasPrice: toUnit(new Big(gasPrice, 16), 'wei', 'gwei').toString(),
-      nonce: new Big(nonce, 16).toString(),
+      fromAddress: '',
       timeToRead: 5
```

Comment:
It's async. I saw your note elsewhere about not making it async, and having worked on the Trezor wallet a bit, I totally see what you mean. All of that should probably get resolved before wallet creation.

But right now, I have to keep it in state because async.